src/pages/add/select-currency/select-currency.ts

```diff
@@ -184,61 +177,31 @@ export class SelectCurrencyPage {
     });
   }
 
-  // Tokens
-  public setWallets(): void {
-    this.wallets = this.navParam.data.keyId
+  public showPairedWalletSelector(token) {
+
+    const eligibleWallets = this.navParam.data.keyId
       ? this.profileProvider.getWalletsFromGroup({
           keyId: this.navParam.data.keyId,
-          network: 'livenet'
+          network: 'livenet',
+          pairFor: token,
         })
       : [];
-    this.walletsEth = this.wallets.filter(wallet => wallet.coin == 'eth');
-  }
-
-  public showWalletSelector(token) {
-    const tokenWalletIds = this.wallets
-      .filter(wallet => wallet.coin === token.symbol.toLowerCase())
-      .map(wallet => wallet.id);
-    const availableWallets = this.wallets.filter(
-      wallet =>
-        !tokenWalletIds.includes(`${wallet.id}-${token.address}`) &&
-        wallet.coin === 'eth'
-    );
 
     const walletSelector = this.actionSheetProvider.createInfoSheet(
       'addTokenWallet',
       {
-        wallets: availableWallets,
+        wallets: eligibleWallets,
         token
       }
     );
     walletSelector.present();
-    walletSelector.onDidDismiss(wallet => {
-      if (!_.isEmpty(wallet)) {
-        this.addTokenWallet(wallet, token);
-        this.endProcess();
+    walletSelector.onDidDismiss(pairedWallet => {
+      if (!_.isEmpty(pairedWallet)) {
+        this.addTokenWallet(pairedWallet, token);
       }
+      this.endProcess();
```

Comment:
`this.endProcess()` needs to be inside the if statement or the app will go back to the home screen on `addTokenWallet` info-sheet dismiss even if you dont select an ETH wallet.